File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool: 
        T=list(t)
        S=list(s)
        if len(t) != len(s) :
            return False
        else :
            for i in T :
                if i in S :
                    S.remove(i)
                else :
                    return False
            if S==[] :
                return True

File: Data Structures & Algorithms/subsets-ii/submission-2.java
class Solution {
    Set<List<Integer>> set = new HashSet<>();
    void dfs(int i ,List<Integer> current,int n, int[] nums){
        List<Integer> copy = new ArrayList<>(current);
        Collections.sort(copy);
        if(i==n){
            set.add(copy);
            return ;
        }
        set.add(copy);
        current.add(nums[i]);
        dfs(i+1,current,n,nums);
        current.remove(current.size()-1);
        dfs(i+1,current,n,nums);
    }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        dfs(0,new ArrayList<>(),nums.length,nums);
        return new ArrayList<List<Integer>>(set);
    }
}
